verifier/testreport.py

```diff
@@ -651,15 +651,22 @@ def characterize_results_by_options(self, test_list, category):
                         'strength', 'caseFirst', 'backwards',
                         'reorder', 'maxVariable',
                         'source_file'
+                        # TODO!!! Characterize by actual_options keys & values
                         ]
             for key in key_list:
                 if test.get(key, None):  # For collation results
                     value = test[key]
+                    if not isinstance(value, str):
+                        # MAke this a string for comparison
+                        value = str(value)
                     if key not in results:
                         results[key] = {}
-                    if value not in results[key]:
-                        results[key][value] = set()
-                    results[key][value].add(label)
+                    try:
+                        if value not in results[key]:
+                            results[key][value] = set()
+                        results[key][value].add(label)
```

Comment:
This can be simplified by only keeping the last line. This works because of the previous change you made to initialize `results` using `defaultdict` like so: `results = defaultdict(lambda : defaultdict(list))`. See code above for an example, [see here](https://realpython.com/python-defaultdict/#understanding-the-python-defaultdict-type) for an explanation.
```suggestion
                        results[key][value].add(label)
```

Reply:
Done